node/src/types/block.rs

```diff
@@ -450,26 +458,30 @@ impl FinalizedBlock {
             let rewards_count = rng.gen_range(0..5);
             let inactive_count = rng.gen_range(0..5);
             Some(EraReport {
-                equivocators: iter::repeat_with(|| PublicKey::from(&SecretKey::ed25519(rng.gen())))
-                    .take(equivocators_count)
-                    .collect(),
+                equivocators: iter::repeat_with(|| {
+                    PublicKey::from(&SecretKey::ed25519_from_bytes(rng.gen::<[u8; 32]>()).unwrap())
+                })
+                .take(equivocators_count)
+                .collect(),
                 rewards: iter::repeat_with(|| {
-                    let pub_key = PublicKey::from(&SecretKey::ed25519(rng.gen()));
+                    let pub_key = PublicKey::from(
+                        &SecretKey::ed25519_from_bytes(rng.gen::<[u8; 32]>()).unwrap(),
+                    );
                     let reward = rng.gen_range(1..(BLOCK_REWARD + 1));
                     (pub_key, reward)
                 })
                 .take(rewards_count)
                 .collect(),
                 inactive_validators: iter::repeat_with(|| {
-                    PublicKey::from(&SecretKey::ed25519(rng.gen()))
+                    PublicKey::from(&SecretKey::ed25519_from_bytes(rng.gen::<[u8; 32]>()).unwrap())
                 })
                 .take(inactive_count)
                 .collect(),
             })
         } else {
             None
         };
-        let secret_key: SecretKey = SecretKey::ed25519(rng.gen());
+        let secret_key: SecretKey = SecretKey::ed25519_from_bytes(rng.gen::<[u8; 32]>()).unwrap();
```

Comment:
What's the point of making the constructor return `Result` if we're not taking advantage of it? Is there going to be a follow-up task to do this?

Reply:
This is just test code so we can just unwrap here I think...

Reply:
It was a random place where I put the comment. I meant the whole PR .